src/SoapCore.Tests/SoapMessageProcessor/Startup.cs

```diff
@@ -0,0 +1,44 @@
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+
+namespace SoapCore.Tests.SoapMessageProcessor
+{
+	public class Startup
+	{
+		public void ConfigureServices(IServiceCollection services)
+		{
+			services.AddSoapCore();
+			services.TryAddSingleton<TestService>();
+			services.AddSoapMessageProcessor(async (message, httpcontext, next) =>
+			{
+				await Task.Delay(1);
+
+				if (httpcontext.Request.Path.Value.Contains("ServiceWithProcessor.svc"))
+				{
+					return Message.CreateMessage(MessageVersion.Soap11, "none");
```

Comment:
I'm getting this error message when I try to `message.WriteBodyContents( <some-writer> )`.

      System.InvalidOperationException: This message cannot support the operation because it has been written.
         at System.ServiceModel.Channels.Message.EnsureWriteMessageState(XmlDictionaryWriter writer)
         at System.ServiceModel.Channels.Message.WriteMessage(XmlDictionaryWriter writer)
         at SoapCore.MessageEncoder.SoapMessageEncoder.WriteMessageAsync(Message message, PipeWriter pipeWriter) in /Users/moisesnessim/code/SoapCore/src/SoapCore/Mess
ageEncoder/SoapMessageEncoder.cs:line 173

Reply:
This is when you are trying to read the message and act differently depending on the contents?
Messages are weird and can only be written (read) once. You need to do something like this

```
var bufferedMessage = message.CreateBufferedCopy(int.MaxValue);
var msg = bufferedMessage.CreateMessage();
var reader = msg.GetReaderAtBodyContents();
var content = reader.ReadInnerXml();

return bufferedMessage.CreateMessage();
```

If everything is in order, return `bufferedMessage.CreateMessage();`. Otherwise create and return your own message.

Reply:
No, this is when I want to return a new Message, but I want it to have some content, like a serialized DTO, or some other XML.
Apparently, even if I return a new Message and not call "next", something down the pipeline will try to write into the new message. 
And, because I already wrote it, the exception is raised.

Reply:
`message.WriteBodyContents( <some-writer> )` writes the content of the message to the supplied writer. Not the other way around.

To create a message with content you have to use one of the overloads of Message.CreateMessage()

I just tried changing my test to:

```
if (httpcontext.Request.Path.Value.Contains("ServiceWithProcessor.svc"))
{
	return Message.CreateMessage(MessageVersion.Soap11, "none", "This is the body");
}
else
{
	return await next(message);
}
```

That worked and gave me 
```
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
  <s:Header>
    <Action s:mustUnderstand="1" xmlns="http://schemas.microsoft.com/ws/2005/05/addressing/none">none</Action>
  </s:Header>
  <s:Body>
    <string xmlns="http://schemas.microsoft.com/2003/10/Serialization/">This is the body</string>
  </s:Body>
</s:Envelope>
```
in return

Reply:
If I'm missing something, please add a code sample of what you're trying to do and I'll see if I can figure it out

Reply:
Oh, thanks! 
I actually forgot that overload existed.
Anyway, because my DTO has attributes for the XmlSerializer, I used [CreateMessage(MessageVersion, String, XmlReader)](https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.channels.message.createmessage?view=dotnet-plat-ext-6.0#System_ServiceModel_Channels_Message_CreateMessage_System_ServiceModel_Channels_MessageVersion_System_String_System_Xml_XmlReader_) , following [this guide](https://weblogs.asp.net/rternier/sending-xml-data-through-wcf-using-system-servicemodel-channels-message-without-using-datacontractserializer) and it worked fine!